### PR TITLE
MINOR: improve code encapsulation between StreamThread and TaskManager

### PR DESCRIPTION
**Pull request by mjsax**
Call for review @guozhangwang @vvcephei

### CODE REVIEW

**Review by abbccdda — APPROVED** [review]
LGTM

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java [hunk 1]
```diff
@@ -86,7 +87,7 @@
     private boolean rebalanceInProgress = false;  // if we are in the middle of a rebalance, it is not safe to commit
 
     // includes assigned & initialized tasks and unassigned tasks we locked temporarily during rebalance
-    private Set<TaskId> lockedTaskDirectories = new HashSet<>();
+    private final Set<TaskId> lockedTaskDirectories = new HashSet<>();
```
Comment (vvcephei) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java:90]: Jinx! https://github.com/apache/kafka/pull/8818#discussion_r436193832

**Review by vvcephei — APPROVED** [review]
Beautiful. Thanks, @mjsax !

**Comment by mjsax** [comment]
Java 8 and Java 11 failed with env errors. Java 14 passed.

**Comment by mjsax** [comment]
Retest this please.

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `2.6`.

**Pull request merged**